### DISCUSSION TITLE
Shared element transition doesnt work with border radius

### DISCUSSION THREAD

**Discussion by jamesheavey** [software-mansion/react-native-reanimated#5729] [Q&A]
https://github.com/software-mansion/react-native-reanimated/assets/76979225/696d621a-756c-4236-af73-8ae95a3943f5

I understand this is an issue that is known, I just want to know if this is being actively worked on? Also does anyone know of any work arounds?

It works as intended if I have the border radius at 0, however when I but it at 100 it breaks and does this.

Please see the start of the video for 100 radius and end for 0

The animated block on both screens looks like this:

```
<Animated.Image
    source={{ uri: selectedEvent.thumbnail }}
    style={{
      width: "100%",
      height: "100%",
      borderRadius: 0, // 100
    }}
    sharedTransitionTag={`event-${selectedEvent.eventId}`}
/>
```
Another issue is the flicker when navigating

**Comment by MatiPl01** [comment]
Hey!

Do you still see this issue? I am trying to reproduce it but the border radius seems to work fine and is animated properly.

https://github.com/software-mansion/react-native-reanimated/assets/52978053/1ffd9b6e-f4da-4831-a09b-719b21a50955

**Reply by jamesheavey** [reply]
Hi yes I am still seeing the issue, i believe it occurs when a circle (border radius 100) transitions to another circle. I think yours works as it is going from rectangle to circle. Something goes wrong with the scaling when both are rad 100

**Reply by MatiPl01** [reply]
Thanks for clarification. I can, indeed, reproduce the issue.
I will pass it to someone who is responsible for layout animations implementation.

https://github.com/software-mansion/react-native-reanimated/assets/52978053/0a8cddca-7522-4d58-a3f8-6b6a6c2fc78e

<details>
  <summary>Code that I used</summary>

```tsx
import * as React from 'react';
import { View, TouchableNativeFeedback, StyleSheet } from 'react-native';
import type {
  NativeStackNavigationProp,
  NativeStackScreenProps,
} from '@react-navigation/native-stack';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import Animated from 'react-native-reanimated';

type ParamList = {
  Screen1?: object;
  Screen2: {
    title: string;
    sharedTransitionTag: string;
  };
};

const photo = require('./SharedElementTransitions/assets/image.jpg');

const Stack = createNativeStackNavigator<ParamList>();

interface CardProps {
  navigation: NativeStackNavigationProp<ParamList>;
  title: string;
  transitionTag: string;
  isOpen?: boolean;
  nextScreen: keyof ParamList;
}

function Card({
  navigation,
  title,
  transitionTag,
  isOpen = false,
  nextScreen,
}: CardProps) {
  const goNext = (screenName: keyof ParamList) => {
    navigation.navigate(screenName, {
      title,
      sharedTransitionTag: transitionTag,
    });
  };

  const size = isOpen ? 300 : 100;

  return (
    <TouchableNativeFeedback
      onPress={() => {
        goNext(nextScreen);
      }}>
      <Animated.Image
        sharedTransitionTag={transitionTag}
        source={photo}
        style={[
          styles.fullWidth,
          { height: size, width: size, borderRadius: 1000 },
        ]}
      />
    </TouchableNativeFeedback>
  );
}

function Screen1({ navigation }: NativeStackScreenProps<ParamList, 'Screen1'>) {
  return (
    <Animated.ScrollView style={styles.flexOne}>
      <Card
        navigation={navigation}
        title={'Title'}
        transitionTag={'tag'}
        nextScreen="Screen2"
      />
    </Animated.ScrollView>
  );
}

function Screen2({
  route,
  navigation,
}: NativeStackScreenProps<ParamList, 'Screen2'>) {
  const { title, sharedTransitionTag } = route.params;

  return (
    <View style={styles.flexOne}>
      <Card
        navigation={navigation}
        title={title}
        transitionTag={sharedTransitionTag}
        isOpen={true}
        nextScreen="Screen1"
      />
    </View>
  );
}

export default function CardExample() {
  return (
    <Stack.Navigator>
      <Stack.Screen
        name="Screen1"
        component={Screen1}
        options={{ headerShown: false }}
      />
      <Stack.Screen
        name="Screen2"
        component={Screen2}
        options={{ headerShown: true }}
      />
    </Stack.Navigator>
  );
}
const styles = StyleSheet.create({
  flexOne: {
    flex: 1,
  },
  fullWidth: {
    width: '100%',
  },
});
```

</details>

**Reply by MatiPl01** [reply]
I talked with @piaskowyk and he confirmed that this is an issue that should be fixed. Could you please open an issue (Bug report) [here](https://github.com/software-mansion/react-native-reanimated/issues/new/choose) and describe the problem? We will look into the possible solution.

**Reply by jamesheavey** [reply]
Hi @MatiPl01 thank you so much for your help. I have created a issue here: https://github.com/software-mansion/react-native-reanimated/issues/6244

Would you be able to add the snack that you made to it?

Thanks again!

**Reply by MatiPl01** [reply]
I added a link to the repo with repro.
What's interesting, I managed to get it work properly when I update the `borderRadius` proportionally to the size of my component.

Let's say, you have component on the first screen with `width` and `height` set to `100` and `borderRadius` also set to `100`.

If your component on the second screen has different dimensions, e.g. `width` and `height` set to `50`, which is the half of initial dimensions, the `borderRadius` should also decrease twice to `50`.

Please let me know if it helps you. 
This is an issue and should be resolved on our side, though, but you can use this workaround for now.

**Reply by jamesheavey** [reply]
In my initial video, the image also flickers when transitioning (you can see it goes black on some transitions) do you happen to know the reason for this? Sometimes it transitions perfectly, othertimes it flickers

**Reply by MatiPl01** [reply]
Please try upgrading reanimated. You mentioned `3.6.1` version in the issue you created. I tested locally with this reanimated version and I also got strange results, such as the complete disappearance of the shared element. After I upgraded to `3.13.0`, I no longer see the issue.

Please check on your side and let me know if it works for you.

| With 3.6.1 | With 3.13.0 |
|-|-|
| <video src="https://github.com/software-mansion/react-native-reanimated/assets/52978053/f18f8f6c-d7fc-481b-8642-d49e493f1c84" /> | <video src="https://github.com/software-mansion/react-native-reanimated/assets/52978053/b13a51b4-022f-41f3-96b4-dcbe5e5803ba" /> |
